src/main/scala/strawman/collection/View.scala

```diff
@@ -154,14 +154,14 @@ object View extends IterableFactory[View] {
     def iterator() = underlying.iterator().flatMap(f)
   }
 
-  /** A view that concatenates elements of the underlying collection with the elements
-   *  of another collection or iterator.
+  /** A view that concatenates elements of the prefix collection or iterator with the elements
+   *  of the suffix collection or iterator.
    */
-  case class Concat[A](underlying: Iterable[A], other: IterableOnce[A]) extends View[A] {
-    def iterator() = underlying.iterator() ++ other
-    override def knownSize = other match {
-      case other: Iterable[_] if underlying.knownSize >= 0 && other.knownSize >= 0 =>
-        underlying.knownSize + other.knownSize
+  case class Concat[A](prefix: IterableOnce[A], suffix: IterableOnce[A]) extends View[A] {
+    def iterator() = prefix.iterator() ++ suffix.iterator()
```

Comment:
Well, only a few constructors are in that situation. We could change the `Concat` constructor to be `collection` private and maybe create another one, public, that would take two `Iterable[A]` as parameters.